cli/io.go

```diff
@@ -386,21 +386,7 @@ func (in *ScriptingCreateInput) Validate() error {
 // BuildScriptingCreateInput constructs a ScriptingCreateInput value.
 func BuildScriptingCreateInput(in options.ScriptingHarness) (*ScriptingCreateInput, error) {
 	out := &ScriptingCreateInput{}
-
-	switch opts := in.(type) {
-	case *options.ScriptingPython:
-		if opts.LegacyPython {
-			out.Type = options.Python2ScriptingType
-		} else {
-			out.Type = options.Python3ScriptingType
-		}
-	case *options.ScriptingGolang:
-		out.Type = options.GolangScriptingType
-	case *options.ScriptingRoswell:
-		out.Type = options.RoswellScriptingType
-	default:
-		return nil, errors.Errorf("unsupported scripting type [%T]", in)
-	}
+	out.Type = in.Type()
```

Comment:
Should we still validate that the type is supported?

Reply:
I assume it's fine because the only way that it would pass an invalid type is if someone defined a new scripting harness type, which is already not a supported behavior. Regardless, if it's an invalid type, it'll fail when the server calls `options.NewScriptingHarness`.